File: src/main/java/com/procurement/access/model/dto/ocds/Option.java
package com.procurement.access.model.dto.ocds;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import org.apache.commons.lang.builder.EqualsBuilder;
import org.apache.commons.lang.builder.HashCodeBuilder;

@Getter
public class Option {

    @JsonProperty("hasOptions")
    private final Boolean hasOptions;

    @JsonCreator
    public Option(@JsonProperty("hasOptions") final Boolean hasOptions) {
        this.hasOptions = hasOptions;
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder()
                .append(hasOptions)
                .toHashCode();
    }

    @Override
    public boolean equals(final Object other) {
        if (other == this) {
            return true;
        }
        if (!(other instanceof Option)) {
            return false;
        }
        final Option rhs = (Option) other;
        return new EqualsBuilder()
                .append(hasOptions, rhs.hasOptions)
                .isEquals();
    }
}

File: src/main/kotlin/com/procurement/access/model/dto/ocds/Option.kt
package com.procurement.access.model.dto.ocds

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.procurement.access.model.dto.databinding.BooleansDeserializer

@JsonInclude(JsonInclude.Include.NON_NULL)
data class Option @JsonCreator constructor(

        @field:JsonDeserialize(using = BooleansDeserializer::class)
        @get:JsonProperty("hasOptions")
        val hasOptions: Boolean?
)